nifi-framework-bundle/nifi-framework/nifi-framework-components/src/main/java/org/apache/nifi/registry/flow/FlowAnalyzingRegistryClientNode.java

```diff
@@ -401,13 +401,33 @@ public boolean isStorageLocationApplicable(final String location) {
     }
 
     @Override
-    public Set<FlowRegistryBucket> getBuckets(final FlowRegistryClientUserContext context) throws FlowRegistryException, IOException {
-        return node.getBuckets(context);
+    public boolean isBranchingSupported() {
+        return node.isBranchingSupported();
     }
 
     @Override
-    public FlowRegistryBucket getBucket(final FlowRegistryClientUserContext context, final String bucketId) throws FlowRegistryException, IOException {
-        return node.getBucket(context, bucketId);
+    public Set<FlowRegistryBranch> getBranches(final FlowRegistryClientUserContext context) throws FlowRegistryException, IOException {
+        return node.getBranches(context);
+    }
+
+    @Override
+    public FlowRegistryBranch getDefaultBranch(final FlowRegistryClientUserContext context) throws FlowRegistryException, IOException {
+        return node.getDefaultBranch(context);
+    }
+
+    @Override
+    public void createBranch(final FlowRegistryClientUserContext context, final CreateBranch createBranch) throws FlowRegistryException, IOException {
```

Comment:
Up to this point, the `FlowAnalyzingRegistryClientNode` did not have methods with argumetns serving as "command object". In cases where it depended on more complex structures, like in case of `RegisteredFlow`, these arguments were data driven (here: `RegisteredFlow` and not `RegisterFlow`). I think this makes the interface into somewhat inconsistent. I would suggest with providing two branch representation, even if they are Strings or their own type (as I see other methods go with String based representation, so if you go with their specific type, I suggest to update those methods as well)

Reply:
The `CreateBranch` object is still just a data driven pojo just like `RegisteredFlow`. I am happy to name it something that sounds less like a command, maybe just adding `Info` to the end - `CreateBranchInfo` ?

I originally had the method as `createBranch(context, String fromBranch, String toBranch)`, but I made the arguments an object following similar thinking to the location objects, meaning - in the future if we have to add another argument, we don't have to break the API signature of `createBranch`, we just add another field to the `CreateBranch` object.

Reply:
Marking this as resolved since createBranch is removed for now.